### ISSUE TITLE
ndpiReader bigendian: wrong packet counters / wrong direction of packets

### ISSUE THREAD

**Issue by vel21ripn** [ntop/nDPI#1479]
## Describe the bug
test pcap/1kxun.pcap failed

### Expected behavior
TCP 192.168.115.8:49613 <-> 183.131.48.144:80 [proto: 7/HTTP][ClearText][Confidence: DPI][cat: Media/1][260 pkts/15070 bytes <-> 159 pkts/168623 bytes]

### Obtained behavior
TCP 192.168.115.8:49613 -> 183.131.48.144:80 [proto: 7/HTTP][ClearText][Confidence: DPI][cat: Media/1][419 pkts/183693 bytes -> 0 pkts/0 bytes]

## nDPI Environment (please complete the following information):
* OS name: Ubuntu
* OS version: 18.04
* Architecture: s390
* gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~18.04)
* libpcap-dev/bionic-updates,now 1.8.1-6ubuntu1.18.04.2 s390x
* ndpi/dev commit 95a3d4fffe699823b1cc9730dd0e4e2827b94845

## How to reproduce the reported bug
```
example/ndpiReader -i tests/pcap/1kxun.pcap | grep ':49613'
```

If i added a bpf filter, then there is no error.
```
example/ndpiReader -i tests/pcap/1kxun.pcap -f "tcp and port 49613" | grep ':49613'
```
Simular errors were found in pcap/skype_no_unknown.pcap, pcap/skype.pcap, pcap/steam.pcap, 
synscan.pcap, pcap/teams.pcap, zoom.pcap

IMHO The error is in ndpiReader, not in the ndpi library.

**Comment by IvanNardi** [comment]
> IMHO The error is in ndpiReader, not in the ndpi library.

I have the same suspicion, but I have not been able to find the core issue, yet

**Comment by vel21ripn** [comment]
I managed to partially localize the error, but because of what it occurs, I still cannot understand. 
```
diff --git a/example/reader_util.c b/example/reader_util.c
index a04f009d..9dc1085e 100644
--- a/example/reader_util.c
+++ b/example/reader_util.c
@@ -876,6 +876,15 @@ static struct ndpi_flow_info *get_ndpi_flow_info(struct ndpi_workflow * workflow
        memset(newflow->ndpi_flow, 0, SIZEOF_FLOW_STRUCT);
 
       ndpi_tsearch(newflow, &workflow->ndpi_flows_root[idx], ndpi_workflow_node_cmp); /* Add */
+      if(ndpi_workflow_node_cmp(newflow,&flow)) {
+             printf("a0 bug\n"); abort();
+      }
+      if(!ndpi_tfind(newflow,&workflow->ndpi_flows_root[idx], ndpi_workflow_node_cmp)) {
+             printf("a1 bug\n"); abort();
+      }
+      if(!ndpi_tfind(&flow,&workflow->ndpi_flows_root[idx], ndpi_workflow_node_cmp)) {
+             printf("a2 bug\n"); abort();
+      }
       workflow->stats.ndpi_flow_count++;
       if(*proto == IPPROTO_TCP)
         workflow->stats.flow_count[0]++;
```
On the bigendian machine I get "a2 bug Aborted" which shouldn't happen.

**Comment by utoni** [comment]
Fixed with PR #1482. Thanks!
